api/security/policy.go

```diff
@@ -0,0 +1,85 @@
+package security
+
+import (
+	"net/http"
+
+	"github.com/gobwas/glob"
+)
+
+// HasOne returns a gin handler that checks the request against the given role
+func (policy RolePolicy) HasOne(roles ...Role) bool {
+	for _, role := range roles {
+		if policy[role] {
+			return true
+		}
+	}
+	return false
+}
+
+// HasAll returns a gin handler that checks the request against the given role
+func (policy RolePolicy) HasAll(roles ...Role) bool {
+	for _, role := range roles {
+		if !policy[role] {
+			return false
+		}
+	}
+	return true
+}
+
+// ToSlice returns a slice of roles
+func (policy RolePolicy) ToSlice() (roles []Role) {
+	for role, ok := range policy {
+		if ok {
+			roles = append(roles, role)
+		}
+	}
+	return
+}
+
+// Compile a policy while compiling values as glob
+func Compile(policy Policy) CompiledPolicy {
```

Comment:
I do not understand the need for a "Compile" function. Why not simply binding/Unmarshalling  to a struct ?

Reply:
Because there is now a *glob* pattern matching to allow syntaxes such as 

```json
{
    "X-Remote-User": ["fabien.*"]
}

Such patterns have to be compiled and its better to do it once at startup instead of doing it at each request.